### PR TITLE
Feat: add scripts to sync archs

### PR DESCRIPTION
**Pull request by maciekstosio**
Changes moved from: https://github.com/software-mansion/react-native-screens/pull/2224

## Description

When changing native props on Fabric, codegen generates corresponding interfaces and delegates. To make sure both implementations are consistent, we implement those interfaces on Paper too. Currently, after generating interfaces using codegen, developer needs to copy corresponding files for paper manually. This task adds Gradle task, that automates this.

## Changes
Current assumption: 
Two scripts: `check-archs-consistency` and `sync-archs`. The first one generates codegen interfaces and compares them with what we have for paper, the second generates and copies for paper to sync the archs.
- sync is run when staged on changes to `src/paper`
- check is run on CI when `src/paper` or `android/src/paper/java/com/facebook/react/viewmanagers` changes

## Test code and steps to reproduce

Open `src/fabric/LineNativeComponent.ts` or/and `src/fabric/NativeSvgRenderableModule.ts` and remove any proper form interface. Now: 
- when building paper, interface should be updated
- when committing, interface should be updated
- if committed and pushed, Test consistency between Paper & Fabric  should fail :)
Brining back the prop and repeating up should cause the interface back and CI green.
Posting changes in other places should cause CI task to run. 

You can also run those commands yourself using `yarn check-archs-consistency` and `yarn sync-archs`

### CODE REVIEW

File: .github/workflows/check-archs-consistency.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Test consistency between Paper & Fabric
+on: 
+  pull_request:
+    branches:
+      - main
+    paths: 
+      - src/fabric/**
+      - android/src/paper/java/com/facebook/react/viewmanagers/**
```
Comment (WoLewicki) [inline .github/workflows/check-archs-consistency.yml:8]: We also have this directory: https://github.com/software-mansion/react-native-svg/tree/2da02cf2593feb8b03489aebfbf992be918d2197/android/src/paper/java/com/horcrux/svg that is autogenerated and should be checked.

File: .github/workflows/check-archs-consistency.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        uses: actions/checkout@v4
+      - name: Use Node.js 18
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: 'yarn'
+      - name: Install node dependencies
+        run: yarn
+      - name: Check Android Paper & Fabric generated interfaces consistency
+        run: yarn check-archs-consistency
```
Comment (WoLewicki) [inline .github/workflows/check-archs-consistency.yml:26]: no newline

File: scripts/codegen-check-consistency.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const { checkCodegenIntegrity } = require('./codegen-utils');
+
+ checkCodegenIntegrity();
```
Comment (WoLewicki) [inline scripts/codegen-check-consistency.js:3]: no newline

File: scripts/codegen-utils.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
+const fs = require('fs');
+const path = require('path');
+const { execSync } = require('child_process');
+const packageJSON = require('../package.json');
+
+const ERROR_PREFIX = 'react-native-gesture-handler';
```
Comment (WoLewicki) [inline scripts/codegen-utils.js:6]: not this lib

File: scripts/codegen-utils.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+const OLD_ARCH_DIR = path.resolve(ANDROID_DIR, 'src/paper');
+const SPECS_DIR = path.resolve(ROOT_DIR, packageJSON.codegenConfig.jsSrcsDir);
+const PACKAGE_NAME = packageJSON.codegenConfig.android.javaPackageName;
+const RN_DIR = path.resolve(ROOT_DIR, 'node_modules/react-native');
+const RN_CODEGEN_DIR = path.resolve(
+  ROOT_DIR,
+  'node_modules/@react-native/codegen'
+);
+
+const SOURCE_FOLDERS = 'java/com/facebook/react/viewmanagers';
```
Comment (WoLewicki) [inline scripts/codegen-utils.js:19]: add the other mentioned dir here too.

**Review by WoLewicki — COMMENTED** [review]
Left some comments. Please answer them

File: scripts/codegen-utils.js (hunk 4 above)
Comment (maciekstosio) [reply scripts/codegen-utils.js]: fixed in: https://github.com/software-mansion/react-native-svg/pull/2357/commits/b20a8c8157363c41523bf15b1ff480bc0ad133e3

File: scripts/codegen-check-consistency.js (hunk 3 above)
Comment (maciekstosio) [reply scripts/codegen-check-consistency.js]: fixed in: https://github.com/software-mansion/react-native-svg/pull/2357/commits/b20a8c8157363c41523bf15b1ff480bc0ad133e3

File: .github/workflows/check-archs-consistency.yml (hunk 2 above)
Comment (maciekstosio) [reply .github/workflows/check-archs-consistency.yml]: fixed in: https://github.com/software-mansion/react-native-svg/pull/2357/commits/b20a8c8157363c41523bf15b1ff480bc0ad133e3

File: scripts/codegen-utils.js (hunk 5 above)
Comment (maciekstosio) [reply scripts/codegen-utils.js]: fixed in: https://github.com/software-mansion/react-native-svg/pull/2357/commits/15900e9c96691ac87f64610e9eeffce09c6135c8

File: .github/workflows/check-archs-consistency.yml (hunk 1 above)
Comment (maciekstosio) [reply .github/workflows/check-archs-consistency.yml:8]: fixed in: https://github.com/software-mansion/react-native-svg/pull/2357/commits/15900e9c96691ac87f64610e9eeffce09c6135c8

File: android/spotless.gradle [hunk 6]
```diff
@@ -3,7 +3,7 @@ apply plugin: 'com.diffplug.spotless'
 
 spotless {
   java {
-    target 'src/main/java/**/*.java', 'src/paper/java/com/horcrux/svg/**/*.java'
```
Comment (maciekstosio) [inline android/spotless.gradle:6]: Those files at `src/paper/java/com/horcrux/svg/` are autogenerated, formatting them messes up checking consistency.

**Review by WoLewicki — APPROVED** [review]
LGTM

**Pull request merged**